plugins/kubernetes/app/models/kubernetes/role_validator.rb

```diff
@@ -226,14 +229,21 @@ def validate_env_values
       @errors << "Env values #{bad.join(', ')} must be strings." if bad.any?
     end
 
-    def validate_prerequisites
+    # samson waits for prerequisites to finish, so only resources that complete can be prerequisites
+    def validate_prerequisites_kinds
       allowed = ["Job", "Pod"]
-      bad = (@elements + templates).select do |e|
-        e.dig(*RoleConfigFile::PREREQUISITE) && !allowed.include?(e[:kind]) && RoleConfigFile.primary?(e)
-      end
-      bad.compact!
-      bad.map! { |e| e[:kind] }
-      @errors << "Elements with type #{bad.join(", ")} cannot be prerequisites." if bad.any?
+      return if map_attributes(RoleConfigFile::PREREQUISITE).compact.empty?
+      return if (map_attributes([:kind]) & allowed).any? || templates.empty?
+
+      @errors << "Prerequisites only support #{allowed.join(', ')}"
+    end
+
+    # only a whole role is supported to be a prerequisites, so prerequisites flag needs to be consistent
+    def validate_prerequisites_consistency
+      used = map_attributes(RoleConfigFile::PREREQUISITE).compact
+      return if (used.empty? || used.size == @elements.size) && used.uniq.size <= 1
```

Comment:
same effect, but easier to read:
```suggestion
      return if used.empty? || (used.size == @elements.size && used.uniq.size == 1)
```